lib/cli/analytics.js

```diff
@@ -46,61 +53,50 @@ var AnalyticsOffMessage  = '\nAnalytics is off. \nIf you would like to turn anal
 
+/*
+    Helper function to determine if cli is a production/dev release
+*/
 
-module.exports = function analytics(argv, callback) {
-    var params = argv._.slice(1);
-
-    if (params.length > 0) {
-        switch(params[0]) {
-            case "on" :
-                // only track it if the value is changing
-                if (insight.optOut !== false) {
-                    // must change it before we use it
-                    insight.optOut = false;
-                    insight.trackEvent({"category":category,
-                                        "action":"set analytics",
-                                        "label":"opt in"});
-                }
-                break;
-            case "off" :
-                // track it only if it changes
-                if (insight.optOut !== true) {
-                    // must use it before we change it
-                    insight.trackEvent({"category":category,
-                                        "action":"set analytics",
-                                        "label":"opt out"});
-                    insight.optOut = true;
-                }
-                break;
-            case "reset" :
-                // this is a secret api, mainly just for testing the prompt messages
-                console.log("resetting it .. ");
-                insight.optOut = undefined;
-                // intentional fallthrough!
-            default:
-                // unrecognized param
-                insight.trackEvent({"category":category,
-                                    "action":"set analytics",
-                                    "label":"unrecognized param"});
-                break;
-        }
+function getDebugFlag() {
+    if(/cordova/.test(category)) {
+        return true;
+    } else {
+        return false;
     }
+}
 
-    if (insight.optOut === undefined) {
+/*
+    Helper function to help build up GELF formatted json
+    TODO: how does GA session vs. user id work?
+    below seems to track user w/ session
+*/
 
-    }
-    else {
-        // somewhat confusing tri-state, it could be true|false|undefined
-        if (insight.optOut) {
-            console.log(AnalyticsOffMessage);
-        }
-        else {
-            console.log(AnalyticsOnMessage);
-        }
-    }
-    callback();
-};
+function basicGELF() {
+    return {
+        "version": "1.1",
+        "host": "cli",
+        "short_message": "",
+        "_userID": insight.clientId,
+        "_appVersion": require('../../package.json').version,
+        "_nodeVersion": process.version,
+        "_platform": os_name(),
+        "_env": getDebugFlag() ? 1 : 0
+    };
+}
 
+/*
+    Helper function to send events to the analytics end point
+*/
+function sendAnalytics(data) {
+    if (!data) return;
+
+    request.post({
+        url: 'https://metrics.phonegap.com/gelf',
+        form: JSON.stringify(data)
+    }, function(err, res, body) {
+        // TODO: do we want to report errors to somewhere???
```

Comment:
I wasn't sure what to do for this in the desktop app either, currently just logging either success/failure (obviously can't do this in the CLI). We should standardize, whether it be nothing or do something.